### PR TITLE
fix(ui): add error to input field

### PR DESCRIPTION
**Pull request by kelwang**
close https://github.com/influxdata/vsflux/issues/80

### CODE REVIEW

File: templates/editConn.js [hunk 1]
```diff
@@ -1,12 +1,12 @@
 const vscode = acquireVsCodeApi();
-let err = [];
+let hasErr = false;
```
Comment (bthesorceror) [inline templates/editConn.js]: this global state is a little out of place, would the logic in this file be better encapsulated with a class?

File: templates/editConn.js [hunk 2]
```diff
@@ -80,19 +69,35 @@ function getData() {
...
+      setErr(wrapper, regexErr);
       return "";
     }
   }
   return v;
 }
+
+function clearErr(wrapper) {
+  wrapper.classList.remove("error");
+  let ele = wrapper.getElementsByClassName("errMsg")[0];
```
Comment (bthesorceror) [inline templates/editConn.js]: We might want to check this element is found

Comment (kelwang) [reply templates/editConn.js]: I think it's safe to without additional check. Failures like this one will block the message being sent, so it's easier for developer to figure out where the nil pointer started.

File: templates/editConn.js [hunk 3]
```diff
@@ -1,98 +1,118 @@
...
-      connToken: data.connToken,
-      connOrg: data.connOrg
+  save() {
+    let self = this;
+    document.getElementById("save").addEventListener("click", () => {
+      let data = self.getData();
+      if (self.hasErr) {
+        return {};
+      }
+      self.vscode.postMessage({
```
Comment (bthesorceror) [inline templates/editConn.js]: this is a bound function, you can just use this here and remove the self variable.

File: templates/editConn.js [hunk 4]
```diff
@@ -1,98 +1,118 @@
...
-      pNode.appendChild(document.createTextNode(ele));
-      errMsgs.appendChild(pNode);
+  testConn() {
+    let self = this;
+    document.getElementById("testConn").addEventListener("click", () => {
+      let data = self.getData();
+      if (self.hasErr) {
+        return {};
+      }
+      self.vscode.postMessage({
```
Comment (bthesorceror) [inline templates/editConn.js]: this is a bound function, you can just use this here and remove the self variable.

**Pull request merged**